pkg/maintenance/helm/helmpatcher.go

```diff
@@ -0,0 +1,252 @@
+package helm
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/blang/semver/v4"
+	"github.com/crossplane-contrib/provider-helm/apis/release/v1beta1"
+	"github.com/go-logr/logr"
+	"github.com/spf13/viper"
+	"github.com/vshn/appcat/v4/pkg/common/jsonpatch"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	// ImageContentType name of the image type in the docker registry API
+	ImageContentType = "image"
+	// ActiveTagStatus indicates that the image is active
+	ActiveTagStatus = "active"
+)
+
+// ImagePatcher is a generic maintenance runner for helm based services
+// It can handle semver images as well as images that don't adhere to semver.
+type ImagePatcher struct {
+	k8sClient         client.Client
+	httpClient        *http.Client
+	log               logr.Logger
+	instanceNamespace string
+}
+
+// NewImagePatcher returns a new ImagePatcher
+// It will patch the image tag of any given helm release.
+func NewImagePatcher(c client.Client, hc *http.Client, l logr.Logger) ImagePatcher {
+	return ImagePatcher{
+		k8sClient:  c,
+		httpClient: hc,
+		log:        l,
+	}
+}
+
+// ValuePath describes a generic path within a release's values.
+// It can generate various different representations of the fields, depending on
+// the use case.
+type ValuePath struct {
+	path []string
+}
+
+// GetJSONPatchPath returns the representation of the path as used in for patching the release
+func (h *ValuePath) GetJSONPatchPath() string {
+	return "/spec/forProvider/values/" + strings.Join(h.path, "/")
+}
+
+// AsSlice will return the path as a slice for the useage in with the unstructured module
+func (h *ValuePath) AsSlice() []string {
+	return h.path
+}
+
+// NewValuePath returns a new ValuePath
+func NewValuePath(path ...string) ValuePath {
+	return ValuePath{
+		path: path,
+	}
+}
+
+// Payload contains the response body of a docker image tags
+type Payload struct {
+	Results []Result `json:"results,omitempty"`
+}
+
+// Result is the object that has details of a docker image tag
+type Result struct {
+	// Name is the tag
+	Name string `json:"name,omitempty"`
+	// TagStatus indicates if the tag is active
+	TagStatus string `json:"tag_status,omitempty"`
+	// ContentType indicates whether this Result is an image
+	ContentType string `json:"content_type,omitempty"`
+}
+
+// VersionComparisonStrategy is a function that actually determines if a given release needs to be updated
+// The returned string should contain the latest version, the boolean determines if the version is actually newer or not
```

Comment:
I'm confused here, which boolean you are referring to?

Reply:
the one I removed after writing the comment and realizing it doesn't provide any benefits 🤪